lib/smack/Prelude.cpp

```diff
@@ -0,0 +1,534 @@
+#include "smack/Prelude.h"
+#include "smack/Naming.h"
+#include "smack/SmackOptions.h"
+#include "smack/Regions.h"
+
+namespace smack {
+
+std::string getIntegerTypeName(int width) {
+  return "i" + std::to_string(width);
+}
+
+std::string getBvTypeName(int width) {
+  return "bv" + std::to_string(width);
+}
+
+std::string makeVarName(int i) {
+  return "i" + std::to_string(i);
+}
+
+std::vector<std::string> makeVarNames(int numVarNames) {
```

Comment:
This function seems unnecessary because first it's only used in `makeVars` and `makeVarExprs` and these functions can be easily expressed using only `makeVarName` and `makeVarExpr`.

Moreover, it also appears to me these two functions can be generalized to something like this,
```C++
template <class T>
std::list<T> makeListOfElems(unsigned num, T (*makeElem)(int)) {
  std::list<T> ret;
  for (unsigned i = 1; i <= num; ++i)
     ret.emplace_back(makeElem(i));
  return ret;
} 
```

Reply:
The only problem with what you propose is that `makeVars` additionally takes a `std::string type` variable. Because of this, I don't see how we can use a single argument function pointer to get the behavior of `makeVars`. So instead of using a function pointer to represent the `makeElem` function, I used another template variable. See [this commit](https://github.com/smackers/smack/pull/410/commits/e06330ca7144f4d3a216e9e037db36669ee89b13) for details.

Reply:
If you pass the closure you created to `makeListOfElems`, it should work without using the additional `U`.

Reply:
According to [this answer](https://stackoverflow.com/questions/28746744/passing-capturing-lambda-as-function-pointer/), capturing lambdas can't be converted to function pointers.

Reply:
I didn't know that. Thank you for sharing this link.